File: GameEngine_IOS/GameEngine/Commons/WaveFront/GenericLoader.swift
import Foundation

/**
 * Contains methods used by the wavefront loaders
 */
open class GenericLoader {
    
    internal static let SPLIT_TOKEN = " ";
    internal static let EMPTY_STRING = "";
    
    /**
     *  Max size of a line in the object file
     */
    internal static let MAX_LINE_LENGTH : Int = 512;
    
    /**
     * Parse two strings and return the equivalent vector2f
     *
     * @param xStr
     *            X component
     * @param yStr
     *            Y component
     *
     * @return The parsed vector
     */
    internal static func parseVector2f(_ xStr : String, yStr : String) -> Vector2f {
        let x : Float = Float(xStr) ?? 0.0
        let y : Float = Float(yStr) ?? 0.0
        
        return Vector2f(x: x, y: y);
    }
    
    
    /**
     * Parse three strings and return the equivalent vector3f
     *
     * @param xStr
     *            X component
     * @param yStr
     *            Y component
     * @param zStr
     *            Z component
     *
     * @return The parsed vector
     */
    internal static func parseVector3f(_ xStr : String, yStr : String, zStr : String) -> Vector3f {
        let x : Float = Float(xStr) ?? 0.0
        let y : Float = Float(yStr) ?? 0.0
        let z : Float = Float(zStr) ?? 0.0
        
        return Vector3f(x: x, y: y, z: z);
    }
    
    /**
     *
     * @param line
     *            The line of the file with all the information
     * @param currentLine
     *            Array of strings with differents parts of line
     *
     * @return A a string defined by in one line
     */
    internal static func parseStringComponent(_ line : String, _ currentLine: [String]) -> String {
        let startIndex : Int = currentLine[0].characters.count
        let index = line.characters.index(line.startIndex, offsetBy: startIndex)
        let fileName = line.substring(from: index)
        return fileName.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines);
    }
}

File: GameEngine_Android/commons/src/main/java/com/dferreira/commons/waveFront/GenericLoader.java
package com.dferreira.commons.waveFront;

import com.dferreira.commons.Vector2f;
import com.dferreira.commons.Vector3f;

/**
 * Contains methods used by the wavefront loaders
 */
class GenericLoader {

    static final String SPLIT_TOKEN = " ";

    /**
     * Parse two strings and return the equivalent vector2f
     *
     * @param xStr X component
     * @param yStr Y component
     * @return The parsed vector
     */
    static Vector2f parseVector2f(String xStr, String yStr) {
        float x = Float.parseFloat(xStr);
        float y = Float.parseFloat(yStr);

        return new Vector2f(x, y);
    }

    /**
     * Parse two strings and return the equivalent vector23
     *
     * @param xStr X component
     * @param yStr Y component
     * @param zStr Z component
     * @return The parsed vector
     */
    static Vector3f parseVector3f(String xStr, String yStr, String zStr) {
        float x = Float.parseFloat(xStr);
        float y = Float.parseFloat(yStr);
        float z = Float.parseFloat(zStr);

        return new Vector3f(x, y, z);
    }

    /**
     * @param line        The line of the file with all the information
     * @param currentLine Array of strings with different parts of line
     * @return A a string defined by in one line
     */
    static String parseStringComponent(String line, String[] currentLine) {
        return line.substring(currentLine[0].length()).trim();
    }
}
